test/cpp/api/nn_utils.cpp

```diff
@@ -67,7 +67,7 @@ TEST_F(NNUtilsTest, ClipGradNorm) {
     auto norm = utils::clip_grad_norm_(l->parameters(), max_norm, norm_type);
     auto norm_after = compute_norm(norm_type);
     ASSERT_FLOAT_EQ(norm, norm_before);
-    ASSERT_FLOAT_EQ(norm_after, max_norm);
+    ASSERT_NEAR(norm_after, max_norm, 1e-6);
```

Comment:
What's going on here?

Reply:
This file cacluates the norm as `pow(p).sum()` whereas `clip_grad_norm` uses `norm(p).pow(p)`. Since `norm` uses a basic running sum, it rounds slightly differently.